File: manager_web/src/api/category.ts
import { apiClient } from './config';

export interface Category {
  _id: string;
  name: string;
  description?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CategoryCreate {
  name: string;
  description?: string;
}

export interface CategoryUpdate {
  name?: string;
  description?: string;
  is_active?: boolean;
}

export const categoryApi = {
  // 获取分类列表
  getCategories: async (skip: number = 0, limit: number = 100, isActive?: boolean): Promise<Category[]> => {
    const params = new URLSearchParams();
    params.append('skip', String(skip));
    params.append('limit', String(limit));
    if (isActive !== undefined) {
      params.append('is_active', String(isActive));
    }
    const response = await apiClient.get<Category[]>(`/categories?${params.toString()}`);
    return response.data;
  },

  // 获取单个分类
  getCategory: async (id: string): Promise<Category> => {
    const response = await apiClient.get<Category>(`/categories/${id}`);
    return response.data;
  },

  // 创建分类
  createCategory: async (data: CategoryCreate): Promise<Category> => {
    const response = await apiClient.post<Category>('/categories', data);
    return response.data;
  },

  // 更新分类
  updateCategory: async (id: string, data: CategoryUpdate): Promise<Category> => {
    const response = await apiClient.put<Category>(`/categories/${id}`, data);
    return response.data;
  },

  // 删除分类
  deleteCategory: async (id: string): Promise<{ message: string; success: boolean }> => {
    const response = await apiClient.delete(`/categories/${id}`);
    return response.data;
  },

  // 获取分类数量
  getCategoryCount: async (isActive?: boolean): Promise<number> => {
    const params = isActive !== undefined ? `?is_active=${isActive}` : '';
    const response = await apiClient.get<{ total: number }>(`/categories/count${params}`);
    return response.data.total;
  },
};

File: server/app/models/category.py
"""
微博账户类别数据模型
用于管理微博账户的类型/分类
"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class Category(BaseModel):
    """微博账户类别模型"""
    id: str = Field(..., alias="_id", description="类别ID，MongoDB自动生成")
    name: str = Field(..., description="类别名称")
    description: Optional[str] = Field(default=None, description="类别描述")
    is_active: bool = Field(default=True, description="是否启用")
    created_at: datetime = Field(default_factory=datetime.utcnow, description="创建时间")
    updated_at: datetime = Field(default_factory=datetime.utcnow, description="更新时间")

    class Config:
        populate_by_name = True


class CategoryCreate(BaseModel):
    """创建类别请求模型"""
    name: str = Field(..., description="类别名称，如: 娱乐、科技、体育等")
    description: Optional[str] = Field(default=None, description="类别描述")


class CategoryUpdate(BaseModel):
    """更新类别请求模型"""
    name: Optional[str] = Field(default=None, description="类别名称")
    description: Optional[str] = Field(default=None, description="类别描述")
    is_active: Optional[bool] = Field(default=None, description="是否启用")


class CategoryResponse(BaseModel):
    """类别响应模型"""
    id: str = Field(..., alias="_id", description="类别ID")
    name: str
    description: Optional[str] = None
    is_active: bool = True
    created_at: datetime
    updated_at: datetime

    class Config:
        populate_by_name = True
